src/libstore/build.cc

```diff
@@ -1629,6 +1629,19 @@ void DerivationGoal::buildDone()
            being valid. */
         registerOutputs();
 
+        if (settings.postBuildHook != "") {
+            printMsg(lvlChatty, format("executing post-build hook '%1%'")
+                % settings.postBuildHook);
+            auto outputPaths = drv->outputPaths();
+            Strings args;
+            for (auto outputPath: outputPaths)
+                args.push_front(outputPath);
+            args.push_front("--");
+            args.push_front(drvPath);
```

Comment:
I used cli arguments because that was the simplest to implement (because the `runProgram2` interface doesn't allow to pass extra environment variables, so it requires extending it a bit) and to be more consistent with the `pre-build-hook`. But environment variables would make the hooks simpler indeed. I'll change that

Reply:
(that being said the hook would then have to parse the `OUTPUT_PATHS` variable to get back a list. But that's probably already better)